driver/src/price_finding/naive_solver.rs

```diff
@@ -4,39 +4,52 @@ use dfusion_core::models::{AccountState, Order, Solution, TOKENS};
 
 use crate::price_finding::error::PriceFindingError;
 
-use super::price_finder_interface::PriceFinding;
+use super::price_finder_interface::{Fee, PriceFinding};
 
 pub enum OrderPairType {
     LhsFullyFilled,
     RhsFullyFilled,
     BothFullyFilled,
 }
 
-fn u128_to_u256(x: u128) -> U256 {
-    U256::from_big_endian(&x.to_be_bytes())
-}
 trait Matchable {
-    fn attracts(&self, other: &Order) -> bool;
+    fn attracts(&self, other: &Order, fee: &Option<Fee>) -> bool;
     fn sufficient_seller_funds(&self, state: &AccountState) -> bool;
-    fn match_compare(&self, other: &Order, state: &AccountState) -> Option<OrderPairType>;
+    fn match_compare(
+        &self,
+        other: &Order,
+        state: &AccountState,
+        fee: &Option<Fee>,
+    ) -> Option<OrderPairType>;
     fn opposite_tokens(&self, other: &Order) -> bool;
     fn have_price_overlap(&self, other: &Order) -> bool;
     fn surplus(&self, buy_price: u128, exec_buy_amount: u128, exec_sell_amount: u128) -> U256;
 }
 
 impl Matchable for Order {
-    fn attracts(&self, other: &Order) -> bool {
+    fn attracts(&self, other: &Order, fee: &Option<Fee>) -> bool {
+        // We can only match orders that touch the fee token
```

Comment:
Is this desirable? How are we going to match orders that don't?

Reply:
Yes, otherwise we won't be able to produce a solution that is accepted by the smart contract. Therefore not matching (and thus potentially creating a more trivial but correct solution is better in my opinion).